stripe/src/main/java/com/stripe/android/paymentsheet/ui/StateAdapter.kt

```diff
@@ -0,0 +1,91 @@
+package com.stripe.android.paymentsheet.ui
+
+import android.content.Context
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ArrayAdapter
+import android.widget.TextView
+import com.stripe.android.R
+
+internal class StateAdapter(
+    context: Context,
+    itemLayoutId: Int = R.layout.country_text_view,
+    private val textViewFactory: (ViewGroup) -> TextView
+) : ArrayAdapter<StateAdapter.State>(context, itemLayoutId) {
+
+    override fun getCount(): Int = STATES.size
+
+    override fun getItem(i: Int): State = STATES[i]
+
+    override fun getItemId(i: Int): Long = getItem(i).hashCode().toLong()
+
+    override fun getView(i: Int, view: View?, viewGroup: ViewGroup): View {
+        return when (view) {
+            is TextView -> view
+            else -> textViewFactory(viewGroup)
+        }.also { stateText ->
+            stateText.text = getItem(i).name
+        }
+    }
+
+    data class State(
+        val code: String,
+        val name: String
+    )
+
+    companion object {
+        val STATES = listOf(
+            State("AL", "Alabama"),
+            State("AK", "Alaska"),
+            State("AZ", "Arizona"),
+            State("AR", "Arkansas"),
+            State("CA", "California"),
+            State("CO", "Colorado"),
+            State("CT", "Connecticut"),
+            State("DE", "Delaware"),
+            State("DC", "District Of Columbia"),
```

Comment:
If DC is on this list, should US territories be present as well?

Eg, I found [this list](https://en.wikipedia.org/wiki/List_of_U.S._state_and_territory_abbreviations) which includes Puerto Rico, Guam, etc.

Reply:
Will tackle the state field in a separate PR